Test/Tests/PushAdminTests.swift

```diff
@@ -110,24 +110,20 @@ class PushAdminTests: XCTestCase {
         rest.internal.storage = MockDeviceStorage()
         let group = DispatchGroup()
 
-        group.enter()
         for device in allDeviceDetails {
+            group.enter()
             rest.push.admin.deviceRegistrations.save(device) { error in
                 assert(error == nil, error?.message ?? "no message")
-                if allDeviceDetails.last == device {
-                    group.leave()
```

Comment:
How it even worked if for `allDeviceDetails.count` enters you have only one leave?

Reply:
Oh I see, enter was out of the loop